mono/metadata/reflection.c

```diff
@@ -7627,10 +7627,91 @@ _mono_reflection_parse_type (char *name, char **endptr, gboolean is_recursed,
 	return 1;
 }
 
+
+/**
+ * mono_identifier_unescape_type_name_chars:
+ * @identifier: the display name of a mono type
+ *
+ * Returns:
+ *  The name in internal form, that is without escaping backslashes.
+ *
+ *  The string is modified in place!
+ */
+char*
+mono_identifier_unescape_type_name_chars(char* identifier)
+{
+	char *w, *r;
+	if (!identifier)
+		return NULL;
+	for (w = r = identifier; *r != 0; r++)
+	{
+		char c = *r;
+		if (c == '\\') {
+			r++;
+			if (*r == 0)
+				break;
+			c = *r;
+		}
+		*w = c;
+		w++;
+	}
+	if (w != r)
+		*w = 0;
+	return identifier;
+}
+
+void mono_identifier_unescape_info (MonoTypeNameParse* info);
+
+static
+void unescape_each_type_argument(void* data, void* user_data)
```

Comment:
Our coding convention is for the return type plus all modifiers to be in the first line. So it should be:
static void 
unescape_each_type_argument(void* data, void* user_data)